### PR TITLE
Split Minmod into limiter and troubled-cell indicator

### PR DESCRIPTION
**Pull request by fmahebert**
## Proposed changes

The Minmod limiter (more precisely, whether the Minmod limiter triggers) is sometimes used to trigger other, more sophisticated, limiters. This PR splits Minmod into two parts: the troubled-cell indicator ("is limiting needed?"), and the the limiter ("yes, so let us limit").

The TCI is tested fairly thoroughly through its use in the Minmod limiter. A future PR will refactor the Minmod tests, by separating them into TCI tests and limiter tests.

Depends on #1406.

### Types of changes:

- [ ] Bugfix
- [ ] New feature

### Component:

- [ ] Code
- [ ] Documentation
- [ ] Build system
- [ ] Continuous integration

### Code review checklist

- [ ] The PR passes all checks, including unit tests, `clang-tidy` and `IWYU`.
  For instructions on how to perform the CI checks locally refer to the [Dev
  guide on the Travis CI](https://spectre-code.org/travis_guide.html).
- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).

### CODE REVIEW

**Comment by fmahebert** [comment]
Labeling as "in progress": there is still a little more work to do here.

**Comment by fmahebert** [comment]
This is now ready to review.

File: src/Evolution/DiscontinuousGalerkin/SlopeLimiters/MinmodType.cpp
Comment (nilsdeppe) [inline src/Evolution/DiscontinuousGalerkin/SlopeLimiters/MinmodType.cpp]: We've been preferring `switch` cases for the reason that then we get compiler warnings if a new value gets added to the enum and isn't handled in the switch case.

File: src/Evolution/DiscontinuousGalerkin/SlopeLimiters/MinmodTci.hpp
Comment (nilsdeppe) [inline src/Evolution/DiscontinuousGalerkin/SlopeLimiters/MinmodTci.hpp:24]: - This should be in the `SlopeLimiters` namespace, I think. So `SlopeLimiters::Minmod_detail`
- Do we want to move this into a `SlopeLimiters::tci` namespace instead? I'm guessing this will need some more generalization to work more generically, but I'd be in favor of starting the TCI generalization sooner rather than later. In that case we should consider the directory structure too. Maybe `Evo/DG/Tci` (we could spell out TroubledCellIndicator too)? Or we could do `Evo/DG/SlopeLimiters/Tci` (in this case I'd prefer not spelling it out in full since it's expected someone doing limiting knows what TCI is, and if not can look at the docs). This second point is for discussion, not a concrete request :)

File: src/Evolution/DiscontinuousGalerkin/SlopeLimiters/MinmodTci.hpp
Comment (nilsdeppe) [inline src/Evolution/DiscontinuousGalerkin/SlopeLimiters/MinmodTci.hpp]: Docs inside a `_detail` namespace won't show up (see point 2 in my comment above)

File: src/Evolution/DiscontinuousGalerkin/SlopeLimiters/MinmodTci.hpp
Comment (nilsdeppe) [inline src/Evolution/DiscontinuousGalerkin/SlopeLimiters/MinmodTci.hpp:34]: Is there a reasonable way to write a factored test for this?

**Review by nilsdeppe — CHANGES_REQUESTED** [review]
Overall looks pretty good :+1:

File: src/Evolution/DiscontinuousGalerkin/SlopeLimiters/MinmodTci.hpp
Comment (fmahebert) [reply src/Evolution/DiscontinuousGalerkin/SlopeLimiters/MinmodTci.hpp:24]: Re the second point: I agree in general. I suggest waiting until after this PR, perhaps until we're adding a second TCI in the mix and start to have a better idea of how things should look.

File: src/Evolution/DiscontinuousGalerkin/SlopeLimiters/MinmodTci.hpp
Comment (fmahebert) [reply src/Evolution/DiscontinuousGalerkin/SlopeLimiters/MinmodTci.hpp]: For now I propose to just make this a non-doxygen comment, it will still be useful in that form.

File: src/Evolution/DiscontinuousGalerkin/SlopeLimiters/MinmodTci.hpp
Comment (fmahebert) [reply src/Evolution/DiscontinuousGalerkin/SlopeLimiters/MinmodTci.hpp:34]: Yes, there is. As I wrote in the PR description: "A future PR will refactor the Minmod tests, by separating them into TCI tests and limiter tests."

The factoring isn't quite done yet, but if you prefer I can finish it up and include it as part of this PR.

**Comment by fmahebert** [comment]
@nilsdeppe Can you see if these commits address your comments, at least until we revisit the organization of our TCIs in more detail?

File: src/Evolution/DiscontinuousGalerkin/SlopeLimiters/MinmodTci.hpp
Comment (nilsdeppe) [reply src/Evolution/DiscontinuousGalerkin/SlopeLimiters/MinmodTci.hpp:24]: Talked at SpECTRE hangout. Summary:
- We both agree this should be done but will hold off once the limiter code for WENO is in.

File: src/Evolution/DiscontinuousGalerkin/SlopeLimiters/MinmodTci.hpp
Comment (nilsdeppe) [reply src/Evolution/DiscontinuousGalerkin/SlopeLimiters/MinmodTci.hpp]: I'm fine leaving it as dox comment, since I expect this code, or some wrapper thing to become as part of a public interface.

**Review by nilsdeppe — COMMENTED** [review]
LGTM :+1: please squash

**Comment by fmahebert** [comment]
Squashed and rebased

**Review by nilsdeppe — DISMISSED** [review]

**Comment by fmahebert** [comment]
Re-rebased after dependency #1406 was merged.

**Review by nilsdeppe — APPROVED** [review]

File: src/Evolution/DiscontinuousGalerkin/SlopeLimiters/MinmodTci.hpp
Comment (kidder) [reply src/Evolution/DiscontinuousGalerkin/SlopeLimiters/MinmodTci.hpp:34]: @nilsdeppe agrees that this can wait for a future PR

**Review by kidder — APPROVED** [review]

File: src/Evolution/DiscontinuousGalerkin/SlopeLimiters/MinmodTci.hpp
Comment (fmahebert) [reply src/Evolution/DiscontinuousGalerkin/SlopeLimiters/MinmodTci.hpp:34]: Tests are factored in PR #1433

**Pull request merged**
